### PR TITLE
Fix/currency api

### LINKED ISSUE #463: Add multiple API providers to convert ZCN to USD
- We need to ensure we have multiple API providers to convert ZCN to USD so we don't hit rate limits on any one provider. Alternatively, we can buy a plan from any one vendor.
- See: https://0chain.slack.com/archives/C02AV6MKT36/p1652775265327279
- Whoever takes this issue also needs to turn `price_in_usd` config in `blobber-values.yaml` in `dev-*` networks to `true`.

### PR DESCRIPTION
**Pull request by cnlangzi**
### Changes
-

### Fixes
- fixed https://github.com/0chain/gosdk/issues/463

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber: https://github.com/0chain/blobber/pull/714
- zwalletcli: https://github.com/0chain/zwalletcli/pull/137
- gosdk:  https://github.com/0chain/gosdk/pull/476
- actions: https://github.com/0chain/actions/pull/51

### CODE REVIEW

File: core/token/quote.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func GetTokenUSDRate(ctx context.Context, symbol string) (float64, error) {
+
+	for _, q := range quotes {
+		r, err := q.GetUSDRate(ctx, symbol)
+		if err == nil {
+			return r, nil
+		}
+		log.Println("token: ", err)
```
Comment (peterlimg) [inline core/token/quote.go]: Please do not log error in gosdk

File: core/token/quote.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	return 0, ErrNoAvailableQuoteQuery
+}
+
+type QuoteQuery interface {
+	GetUSDRate(ctx context.Context, symbol string) (float64, error)
```
Comment (peterlimg) [inline core/token/quote.go]: Make the interface and method private, they will only be used in this package

File: core/token/quote.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package token
```
Comment (peterlimg) [inline core/token/quote.go]: token is a good name, so please do not use it for this exchange rate query package. So perhaps rename to tokenrate.

Also, please rename file name to tokenrate.go, remove the prefix 'quote_' from the other 2 go files.

File: core/token/quote.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+	quotes = []QuoteQuery{
+		&coingeckoQuoteQuery{},
+		createCoinmarketcapQuoteQuery(),
+		//more query services
+	}
+
+}
+
+func GetTokenUSDRate(ctx context.Context, symbol string) (float64, error) {
```
Comment (peterlimg) [inline core/token/quote.go]: Rename to GetUSD() to reduce naming repeating, so for package user, this will be tokenrate.GetUSD()

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: core/token/quote.go (hunk 1 above)
Comment (cnlangzi) [reply core/token/quote.go]: fixed

File: core/token/quote.go (hunk 2 above)
Comment (cnlangzi) [reply core/token/quote.go]: fixed

File: core/token/quote.go (hunk 3 above)
Comment (cnlangzi) [reply core/token/quote.go]: fixed

File: core/token/quote.go (hunk 4 above)
Comment (cnlangzi) [reply core/token/quote.go]: fixed

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/currency_api  |
| zboxcli  | staging  |
| zwalletcli  | fix/currency_api  |
| 0chain  | staging  |
| blobber  | fix/currency_api  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/currency_api  |
| zboxcli  | staging  |
| zwalletcli  | fix/currency_api  |
| 0chain  | staging  |
| blobber  | fix/currency_api  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: core/tokenrate/quote.go [hunk 5]
```diff
@@ -0,0 +1,35 @@
+package tokenrate
+
```
Comment (peterlimg) [inline core/tokenrate/quote.go]: rename the quote.go to tokenrate.go

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: core/tokenrate/quote.go (hunk 5 above)
Comment (cnlangzi) [reply core/tokenrate/quote.go]: fixed

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**